File: tests/Generator/resource/ruby/default/author.rb

# An simple author element with some description
class Author
  attr_accessor :title, :email, :categories, :locations, :origin

  def initialize(title, email, categories, locations, origin)
    @title = title
    @email = email
    @categories = categories
    @locations = locations
    @origin = origin
  end
end

File: tests/Generator/resource/rust/default/author.rs
use serde::{Serialize, Deserialize};
use location::Location;

// An simple author element with some description
#[derive(Serialize, Deserialize)]
pub struct Author {
    #[serde(rename = "title")]
    title: String,

    #[serde(rename = "email")]
    email: Option<String>,

    #[serde(rename = "categories")]
    categories: Option<Vec<String>>,

    #[serde(rename = "locations")]
    locations: Option<Vec<Location>>,

    #[serde(rename = "origin")]
    origin: Option<Location>,

}

